### ISSUE TITLE
Deploy custom trained model.

### ISSUE THREAD

**Issue by meetgandhi123** [facebookresearch/detectron2#2067]
I have trained detectron2 on my custom dataset which consists of images and it's labeled from the open image dataset. The model is been trained and registered.

```
from detectron2.data import DatasetCatalog, MetadataCatalog
for dataset in ["train", "valid"]:
    dataset_name = dataset
    print(f"Registering {dataset_name}")
    DatasetCatalog.register(dataset_name, lambda dataset_name=dataset_name: load_json_labels(dataset_name))
    MetadataCatalog.get(dataset_name).set(thing_classes=subset_classes)
cmaker_fireplace_metadata = MetadataCatalog.get("train")

```
While checking the registered databases I can see my database:

`DatasetCatalog._REGISTERED`

```
'lvis_v1_val': <function detectron2.data.datasets.lvis.register_lvis_instances.<locals>.<lambda>>,
 'train': <function __main__.<lambda>>,
 'valid': <function __main__.<lambda>>,
 'voc_2007_test': <function detectron2.data.datasets.pascal_voc.register_pascal_voc.<locals>.<lambda>>, 
```

The code I used for traininig:

```
from detectron2.engine import DefaultTrainer
from detectron2.config import get_cfg

cfg=get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-Detection/retinanet_R_50_FPN_3x.yaml"))
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-Detection/retinanet_R_50_FPN_3x.yaml")
cfg.DATASETS.TRAIN = ("train",)
cfg.DATASETS.TEST = ("valid",)
cfg.DATALOADER.NUM_WORKERS = 2
cfg.SOLVER.IMS_PER_BATCH = 2
cfg.SOLVER.BASE_LR = 0.00125
cfg.SOLVER.MAX_ITER = 900
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128
cfg.MODEL.RETINANET.NUM_CLASSES = 2
os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
trainer = DefaultTrainer(cfg)
trainer.resume_or_load(resume=False) 
trainer.train()
```

This creates an output folder which consists of  4 files (model_final.path, metric.json, last_checkpoint, and events.out.tfeve....).

I wish to run this model on different google colab notebook, so I downloaded this output folder and uploaded it in another notebook. And wrote the following code:

```
cfg = get_cfg()
cfg.MODEL.WEIGHTS = os.path.join(cfg.OUTPUT_DIR, "/content/output/model_final.pth")
cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.85
cfg.DATASETS.TEST = ("valid",) 
predictor = DefaultPredictor(cfg)
```
 output:
`WARNING [09/25 03:54:49 d2.modeling.backbone.resnet]: ResNet.make_stage(first_stride=) is deprecated!  Use 'stride_per_block' or 'stride' instead.`

```
cmaker_fireplace_metadata = MetadataCatalog.get("train")
img="/content/Capture.JPG"
num_amenities = 2 
img=cv2.imread(img)
outputs = predictor(img)
visualizer = Visualizer(img[:, :, ::-1],
                        metadata=cmaker_fireplace_metadata,
                        scale=0.7)
visualizer = visualizer.draw_instance_predictions(outputs["instances"][:num_amenities].to("cpu"))
cv2_imshow(visualizer.get_image()[:, :, ::-1])

```

But the model is showing random outputs. Basically, my question is how can I run a pre-trained model on a different google collab notebook.
